core/src/main/java/org/jruby/RubyThread.java

```diff
@@ -938,31 +950,28 @@ public boolean isAlive(){
 
     @JRubyMethod
     public IRubyObject fetch(ThreadContext context, IRubyObject key, Block block) {
-        Ruby runtime = context.runtime;
-
-        IRubyObject value = op_aref(key);
+        IRubyObject value = op_aref(context, key);
 
-        if (value.isNil()) {
+        if (value == context.nil) {
```

Comment:
Is this intended as an optimization? isNil is bimorphic, so it should inline everywhere, but context.nil will require a field access every time.

Reply:
Correction...isNil is bimorphic so it *could* inline everywhere. It may not because it's called against more than two different types and some JVMs don't do class hierarchy analysis.

Reply:
generally, have been preferring `==` ... should have written a JMH before assuming

Reply:
```
Benchmark                               Mode  Cnt    Score     Error   Units
NilCheckBenchmark.benchIsNilFalse      thrpt    5  437.404 ±  44.222  ops/us
NilCheckBenchmark.benchIsNilFalseCast  thrpt    5  447.824 ± 109.204  ops/us
NilCheckBenchmark.benchIsNilTrue       thrpt    5  431.957 ±  63.293  ops/us
NilCheckBenchmark.benchIsNilTrueCast   thrpt    5  433.628 ±  57.552  ops/us
NilCheckBenchmark.benchNilCmpFalse     thrpt    5  462.117 ±  43.782  ops/us
NilCheckBenchmark.benchNilCmpTrue      thrpt    5  440.100 ±  35.774  ops/us
```
... seems it doesn't matter much